### PR TITLE
Use path based addressing for S3 when host is localhost or IPv4 or 6 address

### PR DESCRIPTION
**Pull request by andybee**
### We really appreciate your commitment to improving Architect

To maintain a high standard of quality in our releases, before merging every pull request we ask that you've completed the following:

- [x] Forked the repo and created your branch from `master`
- [x] Made sure tests pass (run `npm it` from the repo root)
- [ ] Expanded test coverage related to your changes:
  - [ ] Added and/or updated unit tests (if appropriate)
  - [ ] Added and/or updated integration tests (if appropriate)
- [ ] Updated relevant documentation:
  - [ ] Internal to this repo (e.g. `readme.md`, help docs, inline docs & comments, etc.)
  - [ ] [Architect docs (arc.codes)](https://github.com/architect/arc.codes)
- [ ] Summarized your changes in `changelog.md`
- [ ] Linked to any related issues, PRs, etc. below that may relate to, consume, or necessitate these changes

Please also be sure to completed the CLA (if you haven't already).

Learn more about [contributing to Architect here](https://arc.codes/intro/community).

Thanks again!

### CODE REVIEW

**Comment by andybee** [comment]
Contributing this change I have made locally to allow `aws-lite` to be used with s3rver, localstack etc. where, because I'm using `localhost` (or an IP address), we can't use the preferred method of stuffing the bucket name at the front of the host, because that's either unresolvable (`mybucket.localhost`) or plain broken (`mybucket.127.0.0.1`).

File: plugins/s3/src/lib.mjs [hunk 1]
```diff
@@ -1,7 +1,9 @@
+import { isIPv4, isIPv6 } from 'node:net'
 
 function getHost ({ Bucket }, { region, config }) {
   // Deprecated path-style URLs, still necessary for buckets with periods
-  if (/\./.test(Bucket)) {
+  // (also need to be used when emulating S3 locally, check for localhost and an IPv4 or 6 host)
+  if (config.host === 'localhost' || isIPv4(config.host) || isIPv6(config.host) || /\./.test(Bucket)) {
```
Comment (ryanblock) [inline plugins/s3/src/lib.mjs:6]: Would this not be more efficient as `isIP(config.host)`?

**Comment by ryanblock** [comment]
lgtm!

**Comment by andybee** [comment]
Good shout, hadn't spotted the generic `isIP` function. Swapped that out now.

**Pull request merged**